azurerm/internal/services/iotcentral/tests/resource_arm_iotcentral_application_test.go

```diff
@@ -0,0 +1,233 @@
+				return fmt.Errorf("Bad: IoT Central Application %q (Resource Group  %q) does not exist", appName, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get IoT Central Application:  %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func TestAccAzureRMIoTCentralApplication_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iotcentral_application", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotCentralApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotCentralApplication_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotCentralApplicationExists(data.ResourceName),
+				),
+			},
+			{
+				Config: testAccAzureRMIotCentralApplication_update(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotCentralApplicationExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku", "S1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.ENV", "Test"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMIoTCentralApplication_requiresImportErrorStep(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_iotcentral_application", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotCentralApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotCentralApplication_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotCentralApplicationExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMIotCentralApplication_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_iotcentral_application"),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMIotCentralApplicationDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).IoTCentral.AppsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_iotcentral_application" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name); err != nil {
+			if resp.StatusCode != http.StatusNotFound {
+				return fmt.Errorf("Bad: Get IoT Central Application:  %+v", err)
+			}
+		}
+	}
+	return nil
+}
+
+func testAccAzureRMIotCentralApplication_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[2]d"
```

Comment:
Could we use
```suggestion
  name     = "acctestRG-iotc-%[1]d"
```

the first indexed argument should be 1

Reply:
fixed